### ISSUE TITLE
Fix libconsensus support

### ISSUE THREAD

**Issue by NicolasDorier** [MetacoSA/NBitcoin#378]
https://github.com/MetacoSA/NBitcoin/pull/375 introduced a new version of libconsensus in the tests directly downloaded from bitcoin core site.

Sadly, the interface seems to be broken with 0.15.1+ and need to be fixed. Tests disabled the changes.

@lontivero I think you had additional issues that you raised with Mac and Linux build?

**Comment by lontivero** [comment]
@NicolasDorier yes, please take a look at this PR comment: https://github.com/MetacoSA/NBitcoin/pull/375#issuecomment-373969951

Sadly I don't remember the problem very well but I think it was that the consensus library doesn't accept transactions with flag `StrictEnc` because it is defined as `(1 <<  1)` and that value is not a valid flags. The available flags are:

```c++
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_NONE                = 0,
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_P2SH                = (1U << 0), // evaluate P2SH (BIP16) subscripts
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_DERSIG              = (1U << 2), // enforce strict DER (BIP66) compliance
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_NULLDUMMY           = (1U << 4), // enforce NULLDUMMY (BIP147)
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_CHECKLOCKTIMEVERIFY = (1U << 9), // enable CHECKLOCKTIMEVERIFY (BIP65)
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_CHECKSEQUENCEVERIFY = (1U << 10), // enable CHECKSEQUENCEVERIFY (BIP112)
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_WITNESS             = (1U << 11), // enable WITNESS (BIP141)
    bitcoinconsensus_SCRIPT_FLAGS_VERIFY_ALL                 = bitcoinconsensus_SCRIPT_FLAGS_VERIFY_P2SH | bitcoinconsensus_SCRIPT_FLAGS_VERIFY_DERSIG |
                                                               bitcoinconsensus_SCRIPT_FLAGS_VERIFY_NULLDUMMY | bitcoinconsensus_SCRIPT_FLAGS_VERIFY_CHECKLOCKTIMEVERIFY |
                                                               bitcoinconsensus_SCRIPT_FLAGS_VERIFY_CHECKSEQUENCEVERIFY | bitcoinconsensus_SCRIPT_FLAGS_VERIFY_WITNESS
```

**NOTE:** If well in the [src/script/interpreter.h](https://github.com/bitcoin/bitcoin/blob/master/src/script/interpreter.h) file there is a constant `SCRIPT_VERIFY_STRICTENC = (1U << 1)`, that is not among the values in the [src/script/bitcoinconsensus.h](https://github.com/bitcoin/bitcoin/blob/0c5f67b8e5d9a502c6d321c5e0696bc3e9b4690d/src/script/bitcoinconsensus.h#L51) file (I don't know why). Take into account that this values (`bitcoinconsensus_XXXX_YYYY_ZZZZ`) are the ones used in the flags verification process.

This problem started happening once I updated the bitcoin version from 0.15.x to 0.16.0. The root  cause could be that a new flags validation was added in the [bitcoin commit 5ca8ef299](https://github.com/bitcoin/bitcoin/commit/5ca8ef299)

**Comment by NicolasDorier** [comment]
`StrictEnc == DERGSIG`

**Comment by lontivero** [comment]
(1U << 1) != (1U << 2)

**Comment by NicolasDorier** [comment]
maybe we need to tanslate the bits in the wrapper.

**Comment by lontivero** [comment]
Yes, that's probably the best option.
